.github/workflows/macos.yml

```diff
@@ -71,25 +71,31 @@ jobs:
 
       - run: make
 
+      - name: Set test options for skipped tests
+        run: |
+          set -ex
+          TESTS="$(echo "${{ matrix.skipped_tests }}" | sed 's| |$$/ -n!/|g;s|^|-n!/|;s|$|$$/|')"
+          echo "TESTS=${TESTS}" >> $GITHUB_ENV
+        if: ${{ matrix.test_task == 'check' && matrix.skipped_tests }}
```

Comment:
Maybe this step can be shared in `.github/actions/setup/directories/action.yml` or somewhere?

Reply:
I would agree with your intention. I can understand we want to manage a common logic in a common file such as `.github/actions/**`. I think a refactoring for this part is a challenge. Because I see the following challenges.

* I checked the `.github/actions/setup/directories/action.yml`. It seems the file is used to prepare the initial directories and files to test Ruby. I feel that adding a logic for skipped tests is a little different topic.
* A logic to update the `TESTS` is not only in the step `name: Set test options for skipped tests`, but also the part `TESTS="${TESTS//-n!\//-n/}"` in another steps `name: make skipped tests` too.

If you don't mind, I want to keep the current logic in this part right now, and I want to find a way to manage the logic for the skipped tests in a common file. If I find a way, I will send another PR. This PR is a blocker of the https://github.com/ruby/ruby/pull/9777. So, I think that I want to finish this PR sooner.